File: src/main/java/com/example/ssauc/exception/ApiExceptionHandler.java
package com.example.ssauc.exception;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

import java.util.HashMap;
import java.util.Map;

/**
 * 전역 예외 처리: API 호출 시 발생하는 예외를 JSON 형태로 반환
 * 필요하지 않다면 사용하지 않아도 됨
 */
@RestControllerAdvice
public class ApiExceptionHandler {

    @ExceptionHandler(Exception.class)
    public ResponseEntity<Map<String, String>> handleException(Exception ex) {
        Map<String, String> error = new HashMap<>();
        error.put("error", ex.getMessage());
        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(error);
    }
}

File: src/main/java/com/example/ssauc/exception/ApiExceptionHandler.kt
package com.example.ssauc.exception

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

/**
 * 전역 예외 처리: API 호출 시 발생하는 예외를 JSON 형태로 반환
 * 필요하지 않다면 사용하지 않아도 됨
 */
@RestControllerAdvice
class ApiExceptionHandler {
    @ExceptionHandler(Exception::class)
    fun handleException(ex: Exception): ResponseEntity<Map<String, String?>> {
        val error: MutableMap<String, String?> = HashMap()
        error["error"] = ex.message
        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(error)
    }
}
